litellm/litellm_core_utils/core_helpers.py

```diff
@@ -15,43 +15,66 @@
-        return "content_filter"
-    elif finish_reason == "STOP":  # vertex ai
-        return "stop"
-    elif finish_reason == "end_turn" or finish_reason == "stop_sequence":  # anthropic
-        return "stop"
-    elif finish_reason == "max_tokens":  # anthropic
-        return "length"
-    elif finish_reason == "tool_use":  # anthropic
-        return "tool_calls"
-    elif finish_reason == "content_filtered":
-        return "content_filter"
-    return finish_reason
+def map_finish_reason(finish_reason: str) -> str:
+    """
+    Maps finish reasons from various AI providers (OpenAI, Vertex AI, HuggingFace, Cohere, Anthropic) to a consistent set 
+    of values based on OpenAI's finish reason values.
+
+    Provider-specific values:
+    - OpenAI: 'stop', 'length', 'tool_calls', 'content_filter', 'function_call'
+    - Vertex AI: 'FINISH_REASON_UNSPECIFIED', 'STOP', 'MAX_TOKENS', 'SAFETY', 'RECITATION', 'LANGUAGE', 'OTHER', 'BLOCKLIST', 'PROHIBITED_CONTENT', 'SPII', 'MALFORMED_FUNCTION_CALL'
+    - HuggingFace: 'stop_sequence', 'eos_token', 'max_tokens'
+    - Cohere: 'COMPLETE', 'STOP_SEQUENCE', 'MAX_TOKENS', 'TOOL_CALL', 'ERROR'
+    - Anthropic: 'end_turn', 'max_tokens', 'stop_sequence', 'tool_use'
+
+    Provider-speicific values source:
+     - openai.types.chat.chat_completion.Choice.model_fields['finish_reason']
+     - google.generativeai.protos.Candidate.FinishReason.__members__.keys()
+     - cohere.types.ChatFinishReason
+     - anthropic.types.Message.model_fields['stop_reason']
+    """
+    # Mapping of provider-specific finish reasons to standardized values
+    finish_reason_mapping = {
+        # Normal completion reasons
+        "stop": "stop",
+        "COMPLETE": "stop",
+        "ERROR": "stop",
+        "eos_token": "stop",
+        "stop_sequence": "stop",
+        "STOP_SEQUENCE": "stop",
+        "STOP": "stop",
+        "end_turn": "stop",
+        "OTHER": "stop",
+        "FINISH_REASON_UNSPECIFIED": "stop",
+
+        # Error completions reasons
+        # Mapping to stop since our set of finish reasons doesn't include error
+        "ERROR": "stop",
+        "MALFORMED_FUNCTION_CALL": "stop",
+
+        # Length-related reasons
+        "length": "length",
+        "MAX_TOKENS": "length",
+        "max_tokens": "length",
+        
+        # Tool/function call reasons
+        "tool_calls": "tool_calls",
+        "tool_use": "tool_calls",
+        "function_call": "tool_calls",
+        "TOOL_CALL": "tool_calls",
+        
+        # Content filtering/safety reasons
+        "content_filter": "content_filter",
+        "ERROR_TOXIC": "content_filter",
+        "SAFETY": "content_filter",
+        "RECITATION": "content_filter",
+        "content_filtered": "content_filter",
+        "BLOCKLIST": "content_filter",
+        "PROHIBITED_CONTENT": "content_filter",
+        "SPII": "content_filter",
+    }
+    
+    return finish_reason_mapping.get(finish_reason, finish_reason)
```

Comment:
would this return None if unmapped?

Reply:
This returns the `finish_reason` if unmapped, just like the original definition. 
The second value of `get` provides the default value if the key does not exist

 @krrishdholakia

Reply:
nice - can we add a unit test for this, maybe here - https://github.com/BerriAI/litellm/blob/6a45ee1ef7d40b0e6f38a278713479352ed72f57/tests/local_testing/test_utils.py#L4

if you can share a screenshot of it passing, this pr should be good to merge!

Reply:
Done @krrishdholakia. 
<img width="1470" alt="Screenshot 2024-12-25 at 10 38 08 AM" src="https://github.com/user-attachments/assets/22259981-eab2-4958-ac37-2fada9f7a09a" />